File: Week02/Beginner/amertz/funny.py
def is_funny():
    s = input('Enter a string: ')
    r = s[::-1]
    for i in range(1, len(s)):
        if abs(ord(s[i]) - ord(s[i - 1])) != abs(ord(r[i]) - ord(r[i - 1])):
            return False
    return True

print("Funny" if is_funny() else "Not Funny")

File: Week02/Beginner/aelliott/funny.hs
-- funny.hs
{- determines if a string is "funny" i.e. if the absolute difference in ascii 
   values of adjacent characters reads the same forward and backward -}

module Funny where

import Data.Char
import Data.Function

main :: IO ()
main = interact $ unlines . map funnyLine . lines

funnyLine :: String -> String
funnyLine xs | isFunny xs = "Funny"
	     | True    	  = "Not Funny"

isFunny :: String -> Bool
isFunny xs = isPalindrome $ zipWith ((abs .) . (-) `on` ord) xs (tail xs)

isPalindrome :: Eq a => [a] -> Bool
isPalindrome xs = and $ zipWith (==) xs (take ((length xs + 1) `div` 2) $ reverse xs)
